shell/platform/android/io/flutter/embedding/android/FlutterFragmentActivity.java

```diff
@@ -521,6 +521,7 @@ protected FlutterFragment createFlutterFragment() {
             ? TransparencyMode.opaque
             : TransparencyMode.transparent;
     final boolean shouldDelayFirstAndroidViewDraw = renderMode == RenderMode.surface;
+    final boolean shouldAutomaticallyHandleOnBackPressed = Build.VERSION.SDK_INT >= 33;
```

Comment:
With this change, you'll need to update the implementation of `onBackPressed()` below. On Android 33, the `onBackPressed()` method is still relevant, but it won't behave correctly without a call to `super.onBackPressed()`. 

Alternatively, you could make this condition always true and fully remove the `onBackPressed()` override below. That way you have less conditional behavior.

Reply:
I suppose removing the `onBackPressed()` override could impact behavior for any clients that subclass `FlutterFragmentActivity` and themselves override the method, but given that Android now formally discourages that anyways, that seems like a reasonable breaking change.